Source/DafnyCore/AST/Expressions.cs

```diff
@@ -100,73 +99,13 @@ public abstract class Expression : INode {
   public virtual RangeToken RangeToken {
     get {
       if (rangeToken == null) {
-        if (tok is RangeToken tokAsRange) {
-          rangeToken = tokAsRange;
-        } else {
-          var startTok = tok;
-          var endTok = tok;
-
-          void updateStartEndTok(Expression expression) {
-            if (expression.tok.Filename != tok.Filename || expression.IsImplicit || expression is DefaultValueExpression) {
-              // Ignore any auto-generated expressions.
-            } else {
-              if (expression.StartToken.pos < startTok.pos) {
-                startTok = expression.StartToken;
-              }
-              if (endTok.pos < expression.EndToken.pos) {
-                endTok = expression.EndToken;
-              }
-            }
-          }
-
-          SubExpressions.Iter(updateStartEndTok);
-          if (this is StmtExpr stmtExpr) {
-            stmtExpr.S.SubStatements.Iter(s => s.SubExpressions.Iter(updateStartEndTok));
-          }
-
-
-          foreach (var token in Enumerable.Concat(FormatTokens ?? Enumerable.Empty<IToken>(), OwnedTokens)) {
-            if (token.Filename != tok.Filename) {
-              continue;
-            }
-
-            if (token.pos < startTok.pos) {
-              startTok = token;
-            }
-
-            if (token.pos + token.val.Length > endTok.pos + endTok.val.Length) {
-              endTok = token;
-
-            }
-          }
-
-          if (FormatTokens != null) {
-            foreach (var token in FormatTokens) {
-              if (token.Filename != tok.Filename) {
-                continue;
-              }
-
-              if (token.pos < startTok.pos) {
-                startTok = token;
-              }
-
-              if (token.pos + token.val.Length > endTok.pos + endTok.val.Length) {
-                endTok = token;
-              }
-            }
-          }
-
-          rangeToken = new RangeToken(startTok, endTok);
-        }
+        rangeToken = new RangeToken(StartToken ?? tok, EndToken ?? StartToken ?? tok);
```

Comment:
I'm assuming the least. I love to write code that gracefully not fails. I know this is a trade-off, I would prefer to have proven assumptions written in Dafny, and sometimes I like to put contracts, but I've seen with the language server that we have contracts that sometimes fail because our code is not verified, and this results in crashes that could have been avoided.
Would you still prefer me to write a Contract.assume.(...) or would you be just ok that I document this for developers?